src/parameterizations/lateral/MOM_internal_tides.F90

```diff
@@ -1652,41 +1652,39 @@ subroutine reflect(En, NAngle, CS, G, LB)
                                            ! values should collocate with angle_c [nondim]
   logical, dimension(G%isd:G%ied,G%jsd:G%jed) :: ridge
                                            ! tags of cells with double reflection
+  real, dimension(1:Nangle) :: En_reflected ! Energy reflected [R Z3 T-2 ~> J m-2].
 
   real    :: TwoPi                         ! 2*pi = 6.2831853... [nondim]
   real    :: Angle_size                    ! size of beam wedge [rad]
-  real    :: angle_wall                    ! angle of coast/ridge/shelf wrt equator [rad]
-  real, dimension(1:NAngle) :: angle_i     ! angle of incident ray wrt equator [rad]
-  real    :: angle_r                       ! angle of reflected ray wrt equator [rad]
-  real, dimension(1:Nangle) :: En_reflected
-  integer :: i, j, a, a_r, na
-  !integer :: isd, ied, jsd, jed   ! start and end local indices on data domain
-  !                                ! (values include halos)
+  integer :: angle_wall                    ! angle of coast/ridge/shelf wrt equator [rad]
```

Comment:
From inspection of the code, it looks to me like angle_wall and angle_wall0, and angle_r and angle_r0 all must the same units (probably [nondim], because as integers I don't see how they could have units of [rad]).  Please verify that the documented units of these variables are correct.

It might also be worthwhile to document in a comment that the beams are being reflected into a single discrete angle, and are not splitting into multiple angles to approximate an arbitrary new angle.

Reply:
fixed